### PR TITLE
Better i18n support in uf2dialog and cp_front

### PR DESCRIPTION
**Pull request by bletvaska**
* more strings are now available for translation in `uf2dialog` and `cp_front`

### CODE REVIEW

File: thonny/codeview.py
Comment (aivarannamaa) [inline thonny/codeview.py:251]: Most of the translatable strings contain the old style formatting markers, so I suggest you leave `%s` here as well.

File: thonny/plugins/cpython_frontend/cp_front.py
Comment (aivarannamaa) [inline thonny/plugins/cpython_frontend/cp_front.py:244]: Thanks for noticing this typo!

File: thonny/plugins/micropython/pip_gui.py
Comment (aivarannamaa) [inline thonny/plugins/micropython/pip_gui.py:95]: I suppose you have omitted "See .... for more info" because of the piecewise placement of the whole sentence. You could still make it translatable by translating and splitting like this 

```python
trans = tr("See %s for more info")
prefix = trans.split("%s")[0].strip()
suffix = trans.split("%s")[1].strip()
```

File: thonny/plugins/micropython/pip_gui.py
Comment (aivarannamaa) [inline thonny/plugins/micropython/pip_gui.py:91]: Please remove the word "new" from this text. It has to be removed at some point, but removing it later would invalidate all translations written so far.

File: thonny/plugins/micropython/uf2dialog.py
Comment (aivarannamaa) [inline thonny/plugins/micropython/uf2dialog.py:171]: did you forget `tr`?

File: thonny/plugins/micropython/uf2dialog.py
Comment (aivarannamaa) [inline thonny/plugins/micropython/uf2dialog.py:227]: better move the trailing space outside of the `tr` argument

**Review by aivarannamaa — CHANGES_REQUESTED** [review]
Thank you!

Please see my comments. 

(In the future I recommend to make a proposal before undertaking a nontrivial i18n effort, because I've left some strings untranslatable on purpose, eg. because I expect them to change and don't want to add extra work for dozens of translators just by invalidating it soon after)

**Comment by bletvaska** [comment]
ahoj

and thanks for the reply. initially my goal was to improve slovak translation, but then i found more strings, which are normally visible for the users, untranslatable, so i started digging in the code. meanwhile i've was digging again, but i decided to wait for your answer. so - thanks - i'll write my answers to your comments.

File: thonny/codeview.py
Comment (bletvaska) [reply thonny/codeview.py:251]: Yes, I saw it. My decision was based about my experiences. When I teach Python programming and we talk about string formatting in Python, this is something I am not showing to students, or at least I notice them about not to use this style of formatting. I've seen some articles mentioning about not using this style, and also in the official documentation is mentioned, that using such style can be tricky in some situations. Also, when I was translating Mu editor, they were not using "old style" but new one. Question can be about the mentioning the parameter inside of the brackets, so instead of translating string in the form:

```
"Could not read as {} text.\nYou could try another encoding"
```

will be much more "readable" and "explainable" to have string in format:

```
"Could not read as {encoding} text.\nYou could try another encoding"
```

But the final decision is up to you. So I can put it back to "old style".

File: thonny/plugins/micropython/pip_gui.py
Comment (bletvaska) [reply thonny/plugins/micropython/pip_gui.py:95]: My original idea was just about making "visible strings" translatable without any big effort ;) I can take a closer look (also about how tkinter works).

File: thonny/plugins/micropython/pip_gui.py
Comment (bletvaska) [reply thonny/plugins/micropython/pip_gui.py:91]: makes sense

File: thonny/plugins/micropython/uf2dialog.py
Comment (bletvaska) [reply thonny/plugins/micropython/uf2dialog.py:171]: Hmm... It looks like...

File: thonny/plugins/micropython/uf2dialog.py
Comment (bletvaska) [reply thonny/plugins/micropython/uf2dialog.py:227]: You are right. I was experimenting a bit to make it working. And I think, my decision was based on some strings, which also ends with trailing space. I'll fix it.

**Comment by bletvaska** [comment]
Yes - I know about POEditor. My Slovak translation was imported in there.

**Comment by aivarannamaa** [comment]
I now added your changes manually to current master (ab053975576d57eff72987b19c077ff5cf13e00a). Thanks for your work.

**Pull request closed without merging**
